kernel/src/grant.rs

```diff
@@ -318,7 +664,7 @@ impl<'a, T: Default> ProcessGrant<'a, T> {
     /// panic!()`. See the comment in `access_grant()` for more information.
     pub fn enter<F, R>(self, fun: F) -> R
     where
-        F: FnOnce(&mut GrantMemory<T>) -> R,
+        F: FnOnce(&mut GrantMemory<T>, &mut GrantUpcallTable) -> R,
```

Comment:
Why do we pass a mutable reference to `GrantUpcallTable`? My fear is that users are going to be able to `mem::swap` this potentially, though this probably is (a) not possible because of non-reentrant Grants and (b) not a concern given the passed `GrantUpcallTable` reference is dropped afterwards. Still, if there's no good reason for it to be mutable, it shouldn't be.

Reply:
I agree this is not needed, and pushed a commit changing it to a non-mutable reference